src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -936,8 +938,45 @@ namespace OpenLoco::ObjectManager
     // 0x004796A9
     void updateYearly1()
```

Comment:
playtesting note - level crossings already places on the map do not update past their 'designed year' like streetlights - this code only determines what crossing type is used for *new* crossings.